lib/commands/crafter/add_office.rb

```diff
@@ -25,21 +30,25 @@ def updated_apprentice
     end
 
     def run
-      if Office.available?(@office) && (@crafter || @apprentice)
+      if @office.available? && (@crafter || @apprentice)
         if @crafter
-          @crafter.add_office(@office)
-          Logger.info("#{@office} was added to #{@crafter.user_name} - #{updated_crafter}")
+          @crafter.add_office(@office.location)
+          Logger.info(
+            "#{@office.location} was added to #{@crafter.user_name} - #{updated_crafter}"
+          )
         end
 
         if @apprentice
-          @apprentice.add_office(@office)
-          Logger.info("#{@office} was added to #{@apprentice.user_name} - #{updated_apprentice}")
+          @apprentice.add_office(@office.location)
+          Logger.info(
+            "#{@office.location} was added to #{@apprentice.user_name} - #{updated_apprentice}"
+          )
         end
 
-        "You were added to the office: #{@office}"
+        "You were added to the office: #{@office.location}"
       elsif @crafter || @apprentice
-        Logger.info("#{@crafter.user_name} use a unavailable office: #{@office}")
-        "The office available are: #{Office.list.map(&:capitalize).join(", ")}"
+        Logger.info("#{@crafter.user_name} used an unavailable office: #{@office.location}")
+        "The offices available are: #{OfficePresenter.locations}"
```

Comment:
Couldn't the presentation be inline here? I am not sure of the value provided by the presenter as it is not used anywhere else 
```suggestion
        "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
```

Reply:
yeah it might be me optimising too early :)